src/cpp/cpp_id.cpp

```diff
@@ -63,22 +61,16 @@ void cpp_idt::print_fields(std::ostream &out, unsigned indent) const
   for(unsigned i=0; i<indent; i++) out << ' ';
   out << "  class_identifier=" << class_identifier << '\n';
 
-  for(scope_listt::const_iterator
-      it=secondary_scopes.begin();
-      it!=secondary_scopes.end();
-      it++)
+  for(const auto &s : secondary_scopes)
   {
     for(unsigned i=0; i<indent; i++) out << ' ';
-    out << "  secondary_scope=" << (*it)->identifier << '\n';
+    out << "  secondary_scope=" << s->identifier << '\n';
```

Comment:
We could further simplify this loop by using `std::string(indent, ' ')` in place of the `for` loop above. Applies several times across this file.